diff -up infinipath-psm-3.3-2_g6f42cdb_open/buildflags.mak.noexec infinipath-psm-3.3-2_g6f42cdb_open/buildflags.mak
--- infinipath-psm-3.3-2_g6f42cdb_open/buildflags.mak.noexec	2015-03-12 02:00:03.303902296 -0400
+++ infinipath-psm-3.3-2_g6f42cdb_open/buildflags.mak	2015-03-12 02:00:52.289874555 -0400
@@ -75,7 +75,7 @@ BASECFLAGS += $(BASE_FLAGS) $(if $(MIC:0
 	-Wall $(WERROR) $(if $(MIC:0=),-Wno-unused) -fpic -fPIC -D_GNU_SOURCE \
 	$(if $(filter $(CC),icc),,-funwind-tables) $(if $(PSM_PROFILE:0=),-DPSM_PROFILE) \
 	${IPATH_CFLAGS}
-ASFLAGS += $(BASE_FLAGS) $(if $(MIC:0=),$(if $(filter $(CC),icc),-mmic,-D__MIC__)) -g3 -fpic
+ASFLAGS += $(BASE_FLAGS) $(if $(MIC:0=),$(if $(filter $(CC),icc),-mmic,-D__MIC__)) -g3 -fpic -Wa,--noexecstack
 
 LDFLAGS += $(SCIF_LINK_FLAGS)
 
